python/hidet/ir/primitives/cuda/nccl.py

```diff
@@ -0,0 +1,29 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from hidet.ir.expr import Expr
+from hidet.ir.stmt import BlackBoxStmt
+
+# TODO: we should not put nccl-related types here since hidet.cuda.nccl depends on
+#       the existence of nccl library?
+from hidet.cuda.nccl import NcclDataType, NcclRedOp
+
+
+def all_reduce(comm_id: int, sendbuff: Expr, recvbuff: Expr, count: Expr, dtype: NcclDataType, op: NcclRedOp):
```

Comment:
```suggestion
def all_reduce(comm_id: int, sendbuff: Expr, recvbuff: Expr, count: Expr, dtype: DataType, op: NcclRedOp):
```
Let's pass the `hidet.ir.type.DataType` to this primitive and convert it into NcclDataType inside the primitive function.